frontend/apps/emails/components/EmailContent.tsx

```diff
@@ -353,10 +382,27 @@ function renderInlineTextWithAnnotations(
     } else if (annotation.type === 'Code') {
       annotatedText = `<code>${annotatedText}</code>`
     } else if (annotation.type === 'Link') {
-      annotatedText = `<a href="${annotation.link}" style="color: #346DB7;">${annotatedText}</a>`
+      let href = annotation.link
+      if (href?.startsWith('hm://')) {
+        const unpacked = unpackHmId(href)
+        href = createWebHMUrl(unpacked.type, unpacked.uid, {
+          path: unpacked.path,
+          hostname: unpacked.hostname ?? null,
+        })
+      }
+      annotatedText = `<a href="${href}" style="color: #346DB7;">${annotatedText}</a>`
     } else if (annotation.type === 'Embed') {
       const resolved = resolvedNames?.[annotation.link] || annotation.link
-      annotatedText = `<a href="${annotation.link}" style="color: #008060;">@${resolved}</a>`
+
+      let href = annotation.link
+      if (href?.startsWith('hm://')) {
```

Comment:
you can just try unpacking it right away. then just handle the case where the unpacked ID is null